text/0000-feature-metadata.md

```diff
@@ -0,0 +1,296 @@
+    unless `barred` is also marked `deprecated.
+
+## `public`
+
+`public` is a boolean value that defaults to `true`. It can be thought of as
+`pub` in Rust source files, with the exception of being true by default. If set
+to `false`, Cargo should forbid its use with an error message on any downstream
+crates.
+
+In general, marking a feature `public = false` should make tooling treat the
+feature as non-public API. That includes:
+
+-   The feature should not be accepted by `cargo add --features`
+-   The feature should not be reported from the feature output report of `cargo
+    add`
+-   In the future, `rustdoc` should not document these features unless
+    `--document-private-items` is specified
+-   A future tool like `cargo info` shouldn't display information about these
+    features
+
+There needs to be an escape hatch for this for things like benchmarks - RFC TBD
+on how this works.
+
+## `unstable`
+
+`unstable` is a boolean value that defaults to `false`. It should indicate that
+a feature gates something that is not API-stable, or can only be built using
+`nightly` Rust.
+
+---
+
+Use cases for this information will likely develop with time, but one of the
+simplest applications is for information output with `cargo add`:
+
+```text
+crab@rust foobar % cargo add regex
+    Updating crates.io index
+      Adding regex v1.7.3 to dependencies.
+             Features:
+             + perf             Enables all performance related features
+             + perf-dfa         Enables the use of a lazy DFA for matching
+             + perf-inline      Enables the use of aggressive inlining inside
+                                match routines
+             + perf-literal     Enables the use of literal optimizations for
+                                speeding up matches
+             + std              When enabled, this will cause regex to use the
+                                standard library
+             + unicode          Enables all Unicode features
+             - unstable (U)     Some unstable features
+             - deprecated (D)   Not a real feature, but it could be
+
+    Updating crates.io index
+```
+
+Features like `aho-corasick`, `memchr`, or `use_std` would likely be `public =
+false` since they aren't listed on the crate landing page.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+-   Added complexity to Cargo. Parsing is trivial, but exact implementation
+    details do add test surface area
+-   Added Cargo arguments if escape hatches for `public` or `unstable` are created
+-   Docstrings can be lengthy, adding noise to `Cargo.toml`. This could
+    potentially be solved with the below mentioned `doc-file` key.
+-   `unstable` and `public` uses may not be common enough to be worth including
+-   A markdown parser will eventaully be required to properly parse the `doc`
+    field. Cargo can likely get away without one for basic functionality, such
+    as printing the summary with `cargo add`.
```

Comment:
This is broader than just `cargo`.  In general, a markdown parser is required to fully interpret the field

Reply:
I updated the wording here